File: src/main/java/com/ddd/trainings/idea/domain/TrainingDurationVO.java
package com.ddd.trainings.idea.domain;

import lombok.AllArgsConstructor;
import lombok.Getter;
import org.jmolecules.ddd.annotation.ValueObject;

@ValueObject
@AllArgsConstructor
public class TrainingDurationVO {
  
  @Getter
  private final int days;

  public static TrainingDurationVO from(int days) {
        //TODO: Implement validation of number of days (Value Object?)
        return new TrainingDurationVO(days);
    }

  public static TrainingDurationVO empty() {
    return new TrainingDurationVO(0);
  }

  public boolean isEmpty() {
    return days == 0;
  }
} 

File: src/Ddd/Trainings/Idea/Domain/TrainingDurationVO.cs
namespace Ddd.Trainings.Idea.Domain
{
    public class TrainingDurationVO(int days)
  {
    public int Days { get; } = days;

    public static TrainingDurationVO From(int days)
        {
            //TODO: Implement validation of number of days (Value Object?)
            return new TrainingDurationVO(days);
        }

        public static TrainingDurationVO Empty()
        {
            return new TrainingDurationVO(0);
        }

        public bool IsEmpty()
        {
            return Days == 0;
        }
    }
} 